### PR TITLE
Set Sentry user id to hashed identifier

### PR DESCRIPTION
**Pull request by nkuba**
We calculate the ID of the user by hashing the address. This way we won't store any sensitive information in Sentry.

### CODE REVIEW

File: dapp/src/sentry.test.ts
Comment (kkosiorowska) [inline dapp/src/sentry.test.ts:1]: We keep the tests in separate directories. So I'm wondering whether to move them to the [tests](https://github.com/thesis/acre/tree/main/dapp/src/tests) folder.

Comment (nkuba) [reply dapp/src/sentry.test.ts:1]: I updated it in https://github.com/thesis/acre/pull/906 to consistently keep the test in the root `test` folder with other packages.
I propose we merge this PR and update it in #906.

File: dapp/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "@tanstack/react-query-devtools": "^5.49.2",
     "axios": "^1.6.7",
     "bignumber.js": "^9.1.2",
+    "crypto-js": "^4.2.0",
```
Comment (r-czajkowski) [inline dapp/package.json:38]: Do we need the new lib? I think we can use [ethers](https://docs.ethers.org/v6/api/crypto/#sha256) or [viem](https://viem.sh/docs/utilities/sha256.html).

Comment (nkuba) [reply dapp/package.json]: [679db3f](https://github.com/thesis/acre/pull/904/commits/679db3f05a26befb392b3ab2cdfb77eed39155d7)

**Review by r-czajkowski — APPROVED** [review]

**Pull request merged**